core/domain/user_jobs_one_off.py

```diff
@@ -135,6 +139,97 @@ def reduce(key, stringified_values):
                 subscription_services.subscribe_to_collection(key, item['id'])
 
 
+class UserImpactCalculationOneOffJob(jobs.BaseMapReduceJobManager):
+    """ Calculates each user's impact score, where impact score is defined as:
+    Sum of (
+    ln(playthroughs) * (ratings_scaler) * (average(ratings) - 2.5))
```

Comment:
Ah yep, I noticed while reading through the rest of it.

I think you could just add a note at the end to say that the impact score for an exploration is 0 if either of the two cases holds.